commands/store.go

```diff
@@ -0,0 +1,240 @@
+									--gateway=http://remote-site.com:8080 --lang=python
+									--env=MYVAR=myval`,
+	RunE: runStoreDeploy,
+}
+
+func runStoreList(cmd *cobra.Command, args []string) error {
+	items, err := storeList(storeAddress)
+	if err != nil {
+		return err
+	}
+
+	if len(items) == 0 {
+		fmt.Printf("The store is empty.")
+		return nil
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', 0)
+	fmt.Fprintln(w)
+	fmt.Fprintln(w, "FUNCTION\tDESCRIPTION")
+
+	for _, item := range items {
+		fmt.Fprintf(w, "%s\t%s\n", item.Title, item.Description)
+	}
+
+	fmt.Fprintln(w)
+	w.Flush()
+
+	return nil
+}
+
+func runStoreInspect(cmd *cobra.Command, args []string) error {
+	if len(args) < 1 {
+		return fmt.Errorf("please provide the function name")
+	}
+
+	item, err := findFunction(args[0])
+	if err != nil {
+		return err
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', 0)
+	fmt.Fprintln(w)
+	fmt.Fprintln(w, "FUNCTION\tDESCRIPTION\tIMAGE\tFUNCTION PROCESS\tREPO")
+	fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\n",
+		item.Title,
+		item.Description,
+		item.Image,
+		item.Fprocess,
+		item.RepoURL,
+	)
+
+	fmt.Fprintln(w)
+	w.Flush()
+	return nil
+}
+
+func runStoreDeploy(cmd *cobra.Command, args []string) error {
+	if len(args) < 1 {
+		return fmt.Errorf("please provide the function name")
+	}
+
+	item, err := findFunction(args[0])
+	if err != nil {
+		return err
+	}
+
+	// Add the store environement variables to the provided ones from cmd
+	if item.Environment != nil {
+		for _, env := range item.Environment {
+			envvarOpts = append(envvarOpts, env)
+		}
+	}
+
+	return RunDeploy(
+		args,
+		item.Image,
+		item.Fprocess,
+		item.Name,
+		envvarOpts,
+	)
+}
+
+func storeList(store string) ([]storeItem, error) {
+	var results []storeItem
+
+	store = strings.TrimRight(store, "/")
+
+	timeout := 60 * time.Second
+	client := proxy.MakeHTTPClient(&timeout)
+
+	getRequest, err := http.NewRequest(http.MethodGet, store, nil)
+	if err != nil {
+		return nil, fmt.Errorf("cannot connect to OpenFaaS store on URL: %s", store)
+	}
+
+	res, err := client.Do(getRequest)
+	if err != nil {
+		return nil, fmt.Errorf("cannot connect to OpenFaaS store on URL: %s", store)
+	}
+
+	if res.Body != nil {
+		defer res.Body.Close()
+	}
+
+	switch res.StatusCode {
+	case http.StatusOK:
+
+		bytesOut, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read result from OpenFaaS store on URL: %s", store)
+		}
+		jsonErr := json.Unmarshal(bytesOut, &results)
+		if jsonErr != nil {
+			return nil, fmt.Errorf("cannot parse result from OpenFaaS store on URL: %s\n%s", store, jsonErr.Error())
+		}
+	default:
+		bytesOut, err := ioutil.ReadAll(res.Body)
+		if err == nil {
+			return nil, fmt.Errorf("server returned unexpected status code: %d - %s", res.StatusCode, string(bytesOut))
+		}
+	}
+	return results, nil
+}
+
+func findFunction(functionName string) (storeItem, error) {
+	var item storeItem
+
+	items, err := storeList(storeAddress)
```

Comment:
Might be better to have findFunction hold a single responsibility and pass the result of storeList via a function parameter i.e. `findFunction(functionName string, storeItems)`